### ISSUE TITLE
Be able to use JBoss Swarm version declared in module.ceylon

### ISSUE THREAD

**Issue by DiegoCoronel** [ceylon/ceylon.swarm#2]
Currently the plugin runs version `2016.9` even if we declare in module.ceylon the new version `2016.10.0`.

**Comment by FroMage** [comment]
Yeah, that's an issue, but the plugin depends on a specific version: the one it's compiled against. We don't currently have any mechanism to override that.

**Comment by DiegoCoronel** [comment]
Yeah, but from this plugin release until now theres more 2 versions: 2016.10.0 and 2016.11.0 and all this stuff is too new and the releases have really necessary improvements. Maybe we can release some kind of versioned plugin? ceylon-swarm-2016.9, ceylon-swarm-2016.10.0, etc.. ? I know this will be insane in mid and long time :(

**Comment by FroMage** [comment]
Sure, it's frustrating, but the _plugin_ itself depends on a specific version. When you call `ceylon swarm` it loads the version defined in the `.plugin` file which is `ceylon.swarm/1.3.0` and that one has a dependency on `2016.9`. At this point in our tooling, there are no options to override that dependency before the tool itself is loaded.

One option could be to load the swarm plugin dynamically from the tool, using the same tricks `ceylon test` does. That's going to be dodgy but I can try. Then I can give you an option to specify the version on the command-line, but we can't detect it from the `module.ceylon` since you don't even have to import swarm in there, and often won't.

**Comment by FroMage** [comment]
Remember that you can always invoke the `swarmtool` yourself with the ceylon-generated `.war`, and that will use the swarm version you want.

**Comment by FroMage** [comment]
OK, I got something: I don't depend on the swarmtool module, but use the `RepositoryManager` to download it and then execute it in another process. This lets you set the swarm version with `--swarm-version`. 

It's ugly because it relies on reflection to work in Ceylon 1.3.0 (I did not make the `ArtifactOverrides.classifier` field public), and sorta-relies on dynamic modification of any user or dist overrides to set that classifier since there's no other way to set classifiers yet in our module system.

I just tested it with all Swarm releases so at least it works.
